### PR TITLE
Migrate move gesture handle to `scrollBy` function

### PR DESCRIPTION
**Pull request by thanhdang198**
This pull request includes changes to the `CarMapContainer` and `CarMapRenderer` classes to improve map interaction and rendering. The most important changes include adding a new method to move the map, refactoring the map rendering process, and simplifying the onScroll behavior.

Improvements to map interaction:

* [`car_common/src/main/java/nl/flitsmeister/car_common/CarMapContainer.kt`](diffhunk://#diff-e9c91cc61afecdf8cb57859c758ef10259291a50648d24be55c2dacbccf1e612R36-R39): Added a new `moveBy` method to enable moving the map by specified x and y distances.

Refactoring map rendering process:

* [`car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt`](diffhunk://#diff-52e2d1570a5988cff256b2fb02f8251fa5896968383174bdfc89a716bead6624L84-R93): Refactored the `onSurfaceAvailable` method to use `addOnDidBecomeIdleListener` and `addOnWillStartRenderingFrameListener` for drawing on the surface, replacing the recursive drawing approach.

Simplifying onScroll behavior:

* [`car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt`](diffhunk://#diff-52e2d1570a5988cff256b2fb02f8251fa5896968383174bdfc89a716bead6624L196-R192): Simplified the `onScroll` method by using the new `moveBy` method from `CarMapContainer` instead of handling touch events directly.
* [`car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt`](diffhunk://#diff-52e2d1570a5988cff256b2fb02f8251fa5896968383174bdfc89a716bead6624L240-L248): Removed the `createMotionEvent` method as it is no longer needed for the simplified onScroll behavior.

### CODE REVIEW

File: car_common/src/main/java/nl/flitsmeister/car_common/CarMapContainer.kt [hunk 1]
```diff
@@ -33,6 +33,10 @@ class CarMapContainer(
     var surfaceWidth: Int? = null
     var surfaceHeight: Int? = null
 
+    fun moveBy(x: Float, y: Float) {
+        mapboxMapInstance?.scrollBy(-x, -y, 0  )
```
Comment (frankkienl) [inline car_common/src/main/java/nl/flitsmeister/car_common/CarMapContainer.kt]: moveBy -> scrollBy ?
That could be confusing

File: car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt [hunk 2]
```diff
@@ -193,37 +189,8 @@ class CarMapRenderer(
...
-                        it.x,
-                        it.y
-                    )
-                )
-            }
-            currentTouchPoint = null
-        }
-
+        mapContainer.moveBy(distanceX, distanceY)
+
```
Comment (frankkienl) [inline car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt]: scrollBy -> moveBy ?
That could be confusing

File: car_common/src/main/java/nl/flitsmeister/car_common/CarMapContainer.kt (hunk 1 above)
Comment (thanhdang198) [reply car_common/src/main/java/nl/flitsmeister/car_common/CarMapContainer.kt]: Oh, this is my mistake. I’ll update the function name to match the native implementation. Thank you for your review!

File: car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt (hunk 2 above)
Comment (thanhdang198) [reply car_common/src/main/java/nl/flitsmeister/car_common/CarMapRenderer.kt]: Thank you for your review, I'll update this.

**Comment by thanhdang198** [comment]
Hi @frankkienl, I've updated the PR based on your review. Could you please check if there are any errors?

**Comment by frankkienl** [comment]
Looks good to me! Tested on head unit 👍

**Pull request merged**